python/langsmith/schemas.py

```diff
@@ -128,10 +128,17 @@ class DatasetBase(BaseModel):
     name: str
     description: Optional[str] = None
     data_type: Optional[DataType] = None
+    inputs_schema: Optional[Dict[str, Any]] = Field(
+        None, alias="inputs_schema_definition"
+    )
+    outputs_schema: Optional[Dict[str, Any]] = Field(
+        None, alias="outputs_schema_definition"
+    )
 
     class Config:
         """Configuration class for the schema."""
 
+        allow_population_by_field_name = True
```

Comment:
@hinthornw I had to do some pydantic magic to make all this work. Do we do this in the SDK? I see this pattern in runtree, but I know pydantic stuff is frowned upon other parts of the code base.

Reply:
Maybe we just don't use pydantic in thecreate_dataset method anymore?

Reply:
we'll need to remove it from all dataset related areas then, because we'll need to do conversion on any read/create/etc